spec/helpers/request_methods_spec.rb

```diff
@@ -0,0 +1,188 @@
+      let(:prefix_entities_response) { { 'entities' => { 'id' => 2 } } }
+      let(:faraday_stubs) { Faraday::Adapter::Test::Stubs.new }
+      let(:connection) { Faraday.new { |builder| builder.adapter :test, faraday_stubs } }
+
+      before :each do
+        TestRequester.expects(:client).at_least_once.returns(connection)
+        TestRequester.expects(:endpoint).at_least_once.returns('entities')
+      end
+
+      after :each do
+        faraday_stubs.verify_stubbed_calls
+      end
+
+      context 'without an id parameter' do
+        context 'without a prefix' do
+          it 'is expected to make the request without the prefix' do
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            faraday_stubs.get('/entities/') { [200, {}, entities_response] }
+            response = TestRequester.request('', {}, :get)
+            expect(response).to eq(entities_response)
+          end
+        end
+
+        context 'with a prefix' do
+          it 'is expected to make the request with the prefix' do
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            faraday_stubs.get('/prefix/entities/') { [200, {}, prefix_entities_response] }
+            response = TestRequester.request('', {}, :get, 'prefix')
+            expect(response).to eq(prefix_entities_response)
+          end
+        end
+      end
+
+      context 'with an id parameter' do
+        it 'is expected to make the request with the id included' do
+          # stub.get receives arguments: path, headers, block
+          # The block should be a Array [status, headers, body]
+          faraday_stubs.get('/entities/1/exists') { [200, {}, exists_response] }
+          response = TestRequester.request(':id/exists', { id: 1 }, :get)
+          expect(response).to eq(exists_response)
+        end
+      end
+    end
+
+    context 'when the record was not found' do
+      context 'and or the status is 404' do
+        let!(:faraday_stubs) do
+          Faraday::Adapter::Test::Stubs.new do |stub|
+            # stub.get receives arguments: path, headers, block
+            # The block should be a Array [status, headers, body]
+            stub.get('/entities/1') { [404, {}, {}] }
+          end
+        end
+        let!(:connection) { Faraday.new { |builder| builder.adapter :test, faraday_stubs } }
+
+        before :each do
+          TestRequester.expects(:client).at_least_once.returns(connection)
+          TestRequester.expects(:endpoint).at_least_once.returns('entities')
+        end
+
+        it 'is expected to raise a RecordNotFound error' do
+          expect { TestRequester.request(':id', { id: 1 }, :get) }.to raise_error(Likeno::Errors::RecordNotFound)
+          faraday_stubs.verify_stubbed_calls
+        end
+      end
+
+      context 'and or the response has a NotFound error message' do
```

Comment:
This is mostly copy/paste. Feel free to perform any fixes that you judge
necessary and we review them together tomorrow :)

On Tue, 5 Jan 2016, 11:35 marcheing <notifications@github.com> wrote:

> In spec/helpers/request_methods_spec.rb
> <https://github.com/mezuro/likeno/pull/1#discussion_r48844624>:
>
> > +          end
> > +        end
> > +        let!(:connection) { Faraday.new { |builder| builder.adapter :test, faraday_stubs } }
> > +
> > +        before :each do
> > +          TestRequester.expects(:client).at_least_once.returns(connection)
> > +          TestRequester.expects(:endpoint).at_least_once.returns('entities')
> > +        end
> > +
> > +        it 'is expected to raise a RecordNotFound error' do
> > +          expect { TestRequester.request(':id', { id: 1 }, :get) }.to raise_error(Likeno::Errors::RecordNotFound)
> > +          faraday_stubs.verify_stubbed_calls
> > +        end
> > +      end
> > +
> > +      context 'and or the response has a NotFound error message' do
>
> In this context the status also can (and actually should) be 404.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mezuro/likeno/pull/1/files#r48844624>.
>